### DISCUSSION TITLE
Are witnesses permitted now?

### DISCUSSION THREAD

**Discussion by kpadmasola** [souffle-lang/souffle#2443] [Q&A] [answered]
Hi,

I am new to Souffle and Datalog, and was trying out the examples in [tutorial](https://souffle-lang.github.io/tutorial) page. 

In section https://souffle-lang.github.io/tutorial#witnesses-are-not-permitted, it gives the following example which is not supposed to compile, but when I tried it, it worked and gave the output as `100	c`

```datalog
.decl A(n:number, w:symbol)
A(1, "a"). A(10, "b"). A(100, "c").

.decl MaxA(x: number,w:symbol)
MaxA(y, w) :- y = max x:{A(x, w)}.

.output MaxA
```

Does this mean that the tutorial page is out of date?

**Comment by quentin** [comment, marked as answer]
For min an max the witnesses are permitted in some trivial cases.
Related discussions:
https://github.com/souffle-lang/souffle-lang.github.io/issues/132
https://github.com/souffle-lang/souffle/issues/2135
https://github.com/souffle-lang/souffle/issues/1801
